### PR TITLE
Get hacker data from DB -> fixing the /start-verification command

### PR DESCRIPTION
**Pull request by byronwang93**
# TL;DR
now an admin can call `/start-verification` and verify emails

the order that the verification goes in is as follows:
- first it checks if the email belongs to a hacker (checks in Factotum->InitBotInfo->{discordServerId}->applicants)
- if it does, then set fields `isVerified = true`, `discordId = {discordId}` and the verifiedTimestamp
- otherwise it will check if the email belongs to a user that is ANOTHER role other than hacker (ex. mentor, etc) (checks in Factotum->InitBotInfo->{discordServerId}->otherRoles)
- if it does, then set the above fields
- if this also fails then the email cannot be verified

# Description/Changes made
- created 2 new collections -> `applicants` and `otherRoles`
- the base schema needed for `applicants` is: {email: ${email}, isVerified: false}
- the base schema needed for `otherRoles` is: {email: ${email}, role: "mentor"} (can replace mentor with another role)

## Why
- this allows us to verify emails of hackers AND individuals of other roles (a central place for us to add info for latecomers too)

## How to test
- make sure server is initialized
- create an `applicants` collection and `otherRoles` collection in the `initBotInfo` of your discord server on firebase
- add users accordingly
- run `/start-verification` and enter in any email
- if entered a valid email, check that `isVerified = true`

## Tests done
- tested validating a hacker in the `applicants` collection
- tested validating a mentor in the `otherRoles` collection
- tested validating an email in neither

# Breaking Changes

### CODE REVIEW

**Comment by martincai8** [comment]
dude how are you so good at coding

File: db/firebase/firebaseUtil.js [hunk 1]
```diff
@@ -257,27 +257,38 @@ module.exports = {
...
+        let otherRolesRef = getFactotumDoc().collection('InitBotInfo').doc(guildId).collection('otherRoles').where('email', '==', emailLowerCase).limit(1);
+        let otherRolesUser = (await otherRolesRef.get()).docs[0];
         if (user) {
             let returnTypes = [];
 
             /** @type {FirebaseUser} */
             let data = user.data();
+            if (!data?.isVerified) {
+                data.isVerified = true;
+                data.VerifiedTimestamp = admin.firestore.Timestamp.now();
```
Comment (daniel-panhead) [inline db/firebase/firebaseUtil.js:271]: Is there a reason for the inconsistent casing?

**Review by daniel-panhead — COMMENTED** [review]
Very cool :D Testing verifying applicants/other roles seems to work fine on my end

**Review by daniel-panhead — APPROVED** [review]
If the inconsistent casing isn't an issue then should be fine to merge!

**Comment by daniel-panhead** [comment]
Also is there an easy way for logs to add emails right now? Or will that be something that dev does manually in firebase?

File: db/firebase/firebaseUtil.js (hunk 1 above)
Comment (byronwang93) [reply db/firebase/firebaseUtil.js:271]: ooo i put this check so that if someone's already been verified then it wouldnt reverify them + update the verifiedTimestamp, is that what you're referring to? @daniel-panhead

**Comment by byronwang93** [comment]
@daniel-panhead it'll be in a PR soon but I'm finalizing the `/add-members` command which should make it easier (except right now there isn't a way to mass add)

**Comment by daniel-panhead** [comment]
Got it, should be fine to merge then :D
